### DISCUSSION TITLE
Reuse Layers without consuming extra memory

### DISCUSSION THREAD

**Discussion by LiquidGunay** [ggml-org/llama.cpp#4956] [Ideas]
https://www.reddit.com/r/LocalLLaMA/comments/194zwyc/instant_frankenmerges_with_exllamav2/
Can something similar to this thread be implemented in llama.cpp where you don't need to merge a model and load the the whole thing in RAM if you are just reusing layers from the same model. You can just load the layers from the base model once and do multiple forward passes on specific layers.

**Comment by ggerganov** [comment]
It can be implemented, here is a comment explaining: https://github.com/ggerganov/llama.cpp/issues/4718#issuecomment-1873855226
